dashboard/app/views/levels/_free_response.html.haml

```diff
@@ -1,16 +1,33 @@
-:javascript
-  function getResult() {
-    return {
-      response: $('.response').val(),
-      result: true
-    };
-  }
+- level ||= @level
+- last_attempt ||= @last_attempt
+- index ||= nil
+- number ||= nil
+
+- if index
+  :javascript
+    window['level_#{index}'] = {
```

Comment:
Oh, they'd keep that hidden form continually updated with their latest answer?

I guess that could be good.  I kind of like calling into a JS method just because it makes for an easy way to trace into each level's answer generation code in a central place, but could be convinced.